src/ServiceManager.php

```diff
@@ -412,8 +403,11 @@ public function setInvokableClass($name, $class = null)
      * Specify a factory for a given service name.
      *
      * @param string $name Service name
-     * @param string|callable|Factory\FactoryInterface $factory Factory to which
-     *     to map.
+     * @param string|callable|Factory\FactoryInterface $factory  Factory to which to map.
+     * phpcs:disable Generic.Files.LineLength.TooLong
+     * @psalm-param class-string<Factory\FactoryInterface>|callable(ContainerInterface,string,array<mixed>|null)|Factory\FactoryInterface $factory
+     * phpcs:enable Generic.Files.LineLength.TooLong
```

Comment:
```suggestion
     * @psalm-param class-string<Factory\FactoryInterface>
     *     |callable(ContainerInterface,string,array<mixed>|null)
     *     |Factory\FactoryInterface $factory
```

( Matches what you did on lines 453-454, and eliminates the need for the phpcs directives)

Reply:
When I do this locally, I get a psalm error.